src/robot/dial.ts

```diff
@@ -0,0 +1,122 @@
+import { RobotClient } from './Client';
+
+interface DialDirectConf {
+  authEntity?: string;
+  host: string;
+  locationSecret?: string;
+}
+
+/** Check if a url corresponds to a local connection via heuristic */
+const isLocalConnection = (url: string) => url.includes('.local');
+
+const dialDirect = async (conf: DialDirectConf): Promise<RobotClient> => {
```

Comment:
I think we need to either export these or expose the right options so we can better connect.

Reply:
exported the conf types: https://github.com/viamrobotics/viam-typescript-sdk/pull/30/commits/d2344febb12d134f9da95c4b65439898d787ed06

i think most editors will still give hints even if we don't export, but doesn't hurt